src/quest_manager/models.py

```diff
@@ -439,13 +456,40 @@ def get_available_without_course(self, user):
         return qs.not_in_progress_completed_or_cooldown(user)
 
     def all_drafts(self, user):
-        qs = self.get_queryset().filter(visible_to_students=False)
+        """
+        Return a queryset of draft quests (not visible to students)
+        for staff and allowed TAs only.
+
+        Args:
+            user: The user requesting quests from the Drafts tab
+
+            Returns:
+                QuerySet of draft quests if the user is staff or a TA with editable draft quests, empty otherwise.
+        """
+        qs = self.get_queryset(include_archived=False).filter(visible_to_students=False)
 
         if user.is_staff:
             return qs
         else:  # TA
             return qs.editable(user)
 
+    def all_archived(self, user):
+        """
+        Return a queryset of archived quests for staff users only.
+
+        Args:
+            user: The user requesting archived quests
+
+        Returns:
+            QuerySet of archived quests if user is staff, empty list otherwise
+        """
+        # If the default of include_archived=False make sure this includes archived
+        qs = self.get_queryset().filter(archived=True)
```

Comment:
Is this the problem? you need to use `self.get_queryset(include_archived=True)`
Also not need to re-filter, it's already done in get_queryset()